packages/tasit-sdk/src/TasitSdk.test.js

```diff
@@ -145,7 +170,81 @@ describe("Decentraland", () => {
       );
       await executeOrder.waitForNonceToUpdate();
 
-      await confirmBalances(estate, [buyer.address, seller.address], [1, 0]);
+      await confirmBalances(
+        estate,
+        [buyer.address, seller.address],
+        [1, estateIds.length - 1]
+      );
+    });
+
+    describe("Decentraland tasit app test cases", () => {
+      beforeEach(
+        "create sell orders and remove wallets from contracts",
+        async () => {
+          marketplace.setWallet(seller);
+          for (let assetId of estateIds) {
+            const priceInWei = ONE.toString();
+            const expireAt = Date.now() + duration.years(1);
+            const createOrder = marketplace.createOrder(
+              estate.getAddress(),
+              assetId,
+              priceInWei,
+              expireAt,
+              gasParams
+            );
+            await createOrder.waitForNonceToUpdate();
+          }
+
+          mana.removeWallet();
+          land.removeWallet();
+          estate.removeWallet();
+          marketplace.removeWallet();
+        }
+      );
+
+      it("should list marketplace estates sell orders (without wallet)", async () => {
+        const estates = [];
+        for (let id of estateIds) {
+          estates.push({
+            id,
+            name: await estate.getMetadata(id),
+          });
+        }
+
+        expect(estates[0].name).to.equal("cool estate 0x1");
```

Comment:
Is getting metadata testing the same thing as the next test? Maybe we remove that part from this test and expect something else?

Reply:
I guess if we do more in the other test it's not unnecessary. And it's nice that it loops through multiple.

https://github.com/tasitlabs/TasitSDK/pull/177/files/93ca3a3f25b3e8859d844d26545c377b84cf1599#r252490333

I'd have to look - do they expose any ERC721 enumerable-like functionality where you can get the number of estates from the contract?

Reply:
I'm not sure, I'll look for it.

Reply:
estateContract.totalSupply()

Reply:
Okay, I'll take a look!

Reply:
Oh, I guess what we would need for this test would be something more like a `marketplaceContract.totalSupply()`. And I don't think they have that.

Reply:
The last thing I would check in this test is making sure getEstateSellOrder was non-null for each of them in the orders array.

Reply:
I think that only way to know that is monitoring events.
The contract has no function such as `totalSupply`.

Reply:
Yep, agreed

Reply:
In the future, we could also look at whether OpenSea has better liquidity for Decentraland LAND, etc.

Reply:
(And they might expose different methods in their marketplace contracts)

Reply:
Feel free to resolve this thread once you've read those last two messages. The two tests are now testing distinct things with your latest commits.